### PR TITLE
Feature/dev 25/invite people to space

### PR DESCRIPTION
**Pull request by aminbksi**
https://momentum.nifty.pm/l/_W7FY!D7!e2l

• If user joins the space, or accepts the invite and joins the space, the name of the user is removed from that list with people available for invite

### CODE REVIEW

**Review by PAKovalev — APPROVED** [review]

File: src/core/models/OnlineUsersList/OnlineUsersList.ts [hunk 1]
```diff
@@ -39,6 +39,12 @@ const OnlineUsersList = types
   .views((self) => ({
     get isLoading(): boolean {
       return self.usersRequest.isLoading;
+    },
+    filteredPeople(excludedPeopleIds: string[], currentUser: string): UserProfileModelInterface[] {
+      const users = self.searchQuery ? self.searchedUsers : self.users;
+      return users
+        .filter((user) => !excludedPeopleIds?.includes(user.uuid))
+        .filter((user) => user.uuid !== currentUser);
```
Comment (PAKovalev) [inline src/core/models/OnlineUsersList/OnlineUsersList.ts:47]: Remove second filter by current user.

File: src/ui-kit/organisms/OnlineUsersList/OnlineUsersList.tsx [hunk 2]
```diff
@@ -74,24 +76,11 @@ const OnlineUsersList: React.FC<PropsInterface> = ({
...
-        ));
-    }
-
     const sortedUsers: UserProfileModelInterface[] = [
-      ...(invite ? [] : onlineUsersList.users.filter((user) => user.uuid === profile.uuid)),
-      ...onlineUsersList.users.filter((user) => user.uuid !== profile.uuid)
+      ...(invite || onlineUsersList.searchQuery.length >= SEARCH_MINIMAL_CHARACTER_COUNT
+        ? []
+        : [profile]),
+      ...onlineUsersList.filteredPeople(excludedPeople, profile.uuid)
```
Comment (PAKovalev) [inline src/ui-kit/organisms/OnlineUsersList/OnlineUsersList.tsx:83]: `...onlineUsersList.filteredPeople([...excludedPeople, profile.uuid])`

**Review by PAKovalev — CHANGES_REQUESTED** [review]

File: src/stores/MainStore/models/AgoraStore/AgoraStageModeStore/AgoraStageModeStore.ts [hunk 3]
```diff
@@ -332,6 +332,10 @@ const AgoraStageModeStore = types
   .actions((self) => ({
     addStageModeUser(userId: string) {
       if (self.audience.find((user) => user.uid === userId) || userId === self.userId) {
+        self.audience.push({
+          uid: userId,
+          role: ParticipantRoleEnum.SPEAKER
+        });
```
Comment (Daboo1998) [inline src/stores/MainStore/models/AgoraStore/AgoraStageModeStore/AgoraStageModeStore.ts:338]: No, audience should not contain speakers!

File: src/stores/MainStore/models/AgoraStore/AgoraStore.ts [hunk 4]
```diff
@@ -243,6 +243,11 @@ const AgoraStore = types
         ? self.agoraStageModeStore.numberOfAudienceMembers
         : self.agoraMeetingStore.users.length + 1;
     },
+    get meetingPeopleIds(): string[] {
+      return self.isStageMode
+        ? self.agoraStageModeStore.audience.map((user) => user.uid.toString())
```
Comment (Daboo1998) [inline src/stores/MainStore/models/AgoraStore/AgoraStore.ts:248]: Here I would map both audience and users, and not combine speakers and audience in the audience field of `agoraStageModeStore`

**Review by Daboo1998 — CHANGES_REQUESTED** [review]

**Review by Daboo1998 — APPROVED** [review]

**Pull request merged**